src/main/java/autoreconnect/mixin/DisconnectedScreensMixin.java

```diff
@@ -24,14 +28,15 @@ public DisconnectedScreensMixin(Text text) {
         super(text);
     }
 
+    public ButtonWidget.PressAction pressAction = btn -> AutoReconnect.schedule(() -> MinecraftClient.getInstance().execute(this::manualReconnect), 100, TimeUnit.MILLISECONDS);
+
     @Inject(method = "<init>", at = @At("TAIL"))
-    private void constructor(Screen parent, Text title, Text reason, CallbackInfo info) {
-        reconnectButton = new ButtonWidget(
-            0, 0, 0, 20, // init height here, can't change it later, should always be 20
-            Text.translatable("text.autoreconnect.disconnect.reconnect"),
-            // schedule 100ms later to let the button click sound play
-            btn -> AutoReconnect.schedule(
-                () -> MinecraftClient.getInstance().execute(this::manualReconnect), 100, TimeUnit.MILLISECONDS));
+    public void constructor(Screen parent, Text title, Text reason, CallbackInfo info) {
+
+        Builder reconnectBuilder = ButtonWidget.builder(Text.translatable("text.autoreconnect.disconnect.reconnect"), pressAction);
+        
+        ButtonWidget reconnectButton = reconnectBuilder.dimensions(0, 0, 0, 20).build();
```

Comment:
Set the class property `reconnectButton` by removing the type `ButtonWidget` from this line.

Reply:
I have a doubt here. Why is it necessary to remove the type `ButtonWidget`? The `build` method returns a `ButtonWidget` type

Reply:
See line 24: https://github.com/Bstn1802/AutoReconnect/pull/40/files/3f1e004df36d028d7c887953f7c17074f66a02af#diff-d660488318bdc759f8781deb64ae070aae9cebd5109a41379ba0c94ce93daf48R24
There is the class property `reconnectButton` of the type `ButtonWidget` defined. What you're doing _here_ is creating a block-scoped variable also named `reconnectButton` of the type `ButtonWidged` INSIDE the constructor. So when doing this, the return value of the `reconnectBuilder.build()` method isn't being written into the class property, but in a new variable which is only accessible within the method, in this case the constructor.
That's why you get an error when the other methods in this class (like `init()`, `cancelCountdown()` and`countdownCallback()`) try to reconnect you to the server, because the `reconnectButton` class property is not initialized when they need it.
I don't know which IDE you are using, but it should give you a hint when there's a block-scoped variable in your code which is unused:
![image](https://user-images.githubusercontent.com/1964346/216851650-42c46291-f077-4094-8445-ec6971a4f2e0.png)
When removing the type, you are telling Java to not create a new block-scoped variable but to try to set the value of a already existing variable, which might be a class property or another block-scoped variable declared earlier in the method:
![image](https://user-images.githubusercontent.com/1964346/216851779-edc1a072-f5ce-406c-b52b-c81c09f443df.png)
When you look closely at how the `reconnectButton` property was set in the previous version of the code, you see it was also accessing the class property, because there was no type before the variable name:
![image](https://user-images.githubusercontent.com/1964346/216851893-2da42e88-5b30-44c9-b4b2-7b358ae2be05.png)

Overall, I think you did good work with the update, but this one irritates me, have you ever coded in Java before?

Reply:
Oh my god, I'm so stupid

I didn't notice the fact that `reconnectButton` was already initialized at the top of the class, my bad

Pushed the type removal change